### ISSUE TITLE
Adjust rune blade magic features for when the character hits zero endurance

### ISSUE THREAD

**Issue by phylll** [phylll/mychs-macro-magic#44]
- Firemole: if character hits 0 endurance, the flames go out automatically.
- Whirlwind: same with the defensive whirlwind effect.
- Storeheart: if there is an active earth bond, once the character hits zero endurance, the bond is released.

**Comment by phylll** [comment]
Ok, so there is hopefully a single point where anyone's endurance may be reduced to zero, which is `m3mgdModifyEndurance()`. 

Quick-and-dirty solution:
- add extra logic to `m3mgdModifyEndurance()`: if I carry one of these blades with its magic active, and my endurance hits zero, end the effect and produce the output to that effect.

Elegant and scalable solution:
- Have `m3mgdModifyEndurance()` call a character's event handler on zero endurance (pull the command from a character attribute, call `!mmm chat: !mmm do ${call}`)
- Have rune blade scripts register and deregister such event handlers on activating/deactivating rune magic

**Comment by phylll** [comment]
What could a flexible event handler look like? 

It would need to be 

- tied to the character and the event
- ideally allow for several handlers to be called for the same event
- even more ideally in a configurable order.

Since nested lists are not possible until we get https://github.com/michael-buschbeck/mychs-macro-magic/issues/159, for now we can only do single event handlers per char-event combination, and fixed storage for each. Storage would need to be in an attribute since that can be either tied to a character (sheet) and/or the name dynamically generated (`char.(prefix & charID & suffix)`), neither of which are possible for accessing variables.

So, we could store an attribute `m3mgd_event_endurance_floor` in the character's own sheet. The attribute would contain an MMM code line, usually a function call such as `do runeEffectEarthBond("<charID>",default,default,"releaseByExhaustion")` which could be called dynamically through `!mmm chat: ${line}` once pulled from its attribute.

When an earth bond is created, the creating code would write this code line into the event handler attribute. It could even dynamically generate a function and publish it to the game context, and write its call into the attribute:

```
!mmm chat: !mmm script
!mmm chat: !mmm function __myDynFunc()
!mmm chat: !mmm   chat: Hello World!
!mmm chat: !mmm end function
!mmm chat: !mmm publish to game: __myDynFunc
!mmm chat: !mmm end script
!mmm
!mmm do setattr(charID, eventHandlerAttrLabel, "do __myDynFunc()")
```

When an earth bond is released, the handler attribute would be reset to `""`.

**Comment by phylll** [comment]
Alright, so that was fun thinking about. However, with no ways of validating the attribute and thus opening up to arbitrary code injection, including by accident, I decided to go for a simpler solution.

What I ended up implementing is a simple "stop all active rune blade effects" function that is executed by `m3mgdModifyEndurance()` when it hits zero. Drawback: This comes earlier than the combat scripts' main output. It would be better to delay the effect until after the immediate effects of a hit -- including the fact that the character is reaching zero endurance -- were displayed.
